pkg/reconciler/route/resources/virtual_service.go

```diff
@@ -420,38 +459,44 @@ func normalizeRouteWeights(appWeights RouteBindingSlice) RouteBindingSlice {
 
 // buildPathMatchers creates a regex matcher for a given route path.
 // These matcher is used in the virtual service to determine which path a request was sent to
-func buildPathMatchers(rsf v1alpha1.RouteSpecFields) (*istio.HTTPMatchRequest, error) {
+func (hb *httpRoutesBuilder) buildPathMatchers(rsf v1alpha1.RouteSpecFields) (*istio.HTTPMatchRequest, error) {
 	path := path.Join("/", rsf.Path, "/")
 	regexpPath, err := v1alpha1.BuildPathRegexp(path)
 	if err != nil {
 		return nil, fmt.Errorf("failed to convert path to regexp: %s", err)
 	}
 
-	var authorityMatch *istio.StringMatch
-
-	if !strings.HasPrefix(rsf.Host(), "*") {
-		authorityMatch = &istio.StringMatch{
-			MatchType: &istio.StringMatch_Exact{
-				Exact: rsf.Host(),
-			},
-		}
-	}
-
-	return &istio.HTTPMatchRequest{
+	httpMatchRequest := &istio.HTTPMatchRequest{
 		Uri: &istio.StringMatch{
 			MatchType: &istio.StringMatch_Regex{
 				Regex: regexpPath,
 			},
 		},
-		Authority: authorityMatch,
-	}, nil
+	}
+
+	if strings.HasPrefix(rsf.Host(), "*") {
+		return httpMatchRequest, nil
+	}
+
+	httpMatchRequest.Authority = &istio.StringMatch{}
+	if hb.defaultConfig != nil && hb.defaultConfig.RouteHostIgnoringPort {
```

Comment:
Correct, it would function as though the feature was disabled. 
I added a note in the doc for the feature flag noting that the feature will work properly with istio v1.15+